<center><h1>Taller 3 - Solución</h1></center>

In [1]:
import numpy as np
import pandas as pd

1.Considere le siguiente distribución de frecuencias:

a)  Completar la tabla.

In [2]:
li = np.array([0, 10, 20, 30, 100])
ls = np.array([10, 20, 30, 100, 200])

yj = (li+ls)/2
n = 200
nj = np.zeros(len(yj))
nj[0] = 60
nj[1] = n*0.4
nj[2] = 30
nj[3] = n*0.1
nj[4] = n-sum(nj[:len(nj)-1])
hj = nj/n
Nj = nj.cumsum()
Hj = hj.cumsum()

nj = np.append(nj, [sum(nj)])
hj = np.append(hj, [1])
yj = np.append(yj, [sum(yj)])

table = pd.DataFrame([yj, nj, hj, Nj, Hj]).T
table.columns = ['Marca de clase', 'F. Absoluta', 'F. Relativa', 'F. Absoluta Acum', 'F. Relativa Acum']
table.index = ["[0;10]", "(10;20]", "(20;30]", "(30;100]", "(100;200]", "Total"]
print(table.to_string())

           Marca de clase  F. Absoluta  F. Relativa  F. Absoluta Acum  F. Relativa Acum
[0;10]                5.0         60.0         0.30              60.0              0.30
(10;20]              15.0         80.0         0.40             140.0              0.70
(20;30]              25.0         30.0         0.15             170.0              0.85
(30;100]             65.0         20.0         0.10             190.0              0.95
(100;200]           150.0         10.0         0.05             200.0              1.00
Total               260.0        200.0         1.00               NaN               NaN


b) ¿Cuántos individuos asumen un valor de la variable de hasta 100? $\\$
	- 190 Estudiantes asumen un valor de la variable de hasta 100. $\\$
c) ¿Qué porcentaje de individuos asumen un valor de la variable de entre 10 y 100? $\\$
	- 65% de los individuos asumen un valor de la variable de entre 10 y 100. $\\$
e) Calcular el promedio de la variable. $\\$

In [3]:
mean = sum(hj*yj)
mean

285.25

2.Considere el conjunto de datos dados en el archivo de texto taller_03_datos.txt. Este archivo contiene las calificaciones promedio de una muestra aleatoria de estudiantes de una Universidad en el primer semestre de este año. A continuación se presenta el procedimiento importar este conjunto de datos en R.

In [4]:
# Importar base de datos
data = pd.read_csv('taller_03_datos.txt', sep='\t', header=None)[0].to_numpy()
n = len(data)

a) Calcular el máximo, el mínimo, el rango, y el promedio de las calificaciones.

In [5]:
# Maximo
maximo = data.max()
minimo = data.min()
R = round(maximo - minimo, 3)
promedio = data.mean()
print(f'Maximo: {maximo}\nMinimo: {minimo}\nRango: {R}\nPromedio: {promedio}')

Maximo: 4.08
Minimo: 3.4
Rango: 0.68
Promedio: 3.69899


b) Hacer la distribución de frecuencias de las calificaciones con m=10 intervalos.

In [6]:
# Intervalos
m = 10
# Amplitud
a = R/m
# Limites
lim = np.array(list(map(lambda x: (x*a)+min(data), np.arange(m+1))))
nj = pd.cut(sorted(data), lim, include_lowest=True).value_counts()
hj = nj/n
Nj = nj.cumsum()
Hj = hj.cumsum()

table = pd.DataFrame([nj, hj, Nj, Hj]).T
table.columns = ['F. Absoluta', 'F. Relativa', 'F. Absoluta Acum', 'F. Relativa Acum']
print(table)

                F. Absoluta  F. Relativa  F. Absoluta Acum  F. Relativa Acum
(3.399, 3.468]         15.0        0.015              15.0             0.015
(3.468, 3.536]         42.0        0.042              57.0             0.057
(3.536, 3.604]        126.0        0.126             183.0             0.183
(3.604, 3.672]        231.0        0.231             414.0             0.414
(3.672, 3.74]         216.0        0.216             630.0             0.630
(3.74, 3.808]         223.0        0.223             853.0             0.853
(3.808, 3.876]         97.0        0.097             950.0             0.950
(3.876, 3.944]         45.0        0.045             995.0             0.995
(3.944, 4.012]          4.0        0.004             999.0             0.999
(4.012, 4.08]           1.0        0.001            1000.0             1.000


3.Se estudiaron dos métodos de recuperación de proteínas. Se hicieron trece corridas usando cada método y se registró la parte de la proteína recuperada para cada corrida. Los resultados son los siguientes: Método 1: $0.32, 0.35, 0.37, 0.39, 0.42, 0.47, 0.51, 0.58, 0.60, 0.62, 0.65, 0.68, 0.75. Método 2: 0.25, 0.40, 0.48, 0.55, 0.56, 0.58, 0.60, 0.65, 0.70, 0.76, 0.80, 0.91, 0.99$. Calcular el mínimo, el máximo, el rango, y el promedio de cada método. ¿Qué diferencias se observan entre los resultados de los dos métodos?

In [7]:
x_1 = np.array([0.32, 0.35, 0.37, 0.39, 0.42, 0.47, 0.51, 0.58, 0.60, 0.62, 0.65, 0.68, 0.75])
x_2 = np.array([0.25, 0.40, 0.48, 0.55, 0.56, 0.58, 0.60, 0.65, 0.70, 0.76, 0.80, 0.91, 0.99])
# Cálculos del primer método
min_1 = x_1.min()
max_1 = x_1.max()
R_1 = round(max_1 - min_1, 3)
promedio_1 = x_1.mean()
# Cálculos del segundo método
min_2 = x_2.min()
max_2 = x_2.max()
R_2 = round(max_2 - min_2, 3)
promedio_2 = x_2.mean()
# Mostrar los resultados
print(f'Maximo: {max_1}\nMinimo: {min_1}\nRango: {R_1}\nPromedio: {promedio_1}')
print('-'*50)
print(f'Maximo: {max_2}\nMinimo: {min_2}\nRango: {R_2}\nPromedio: {promedio_2}')

Maximo: 0.75
Minimo: 0.32
Rango: 0.43
Promedio: 0.5161538461538462
--------------------------------------------------
Maximo: 0.99
Minimo: 0.25
Rango: 0.74
Promedio: 0.6330769230769231


4.Una muestra de temperaturas para iniciar una cierta reacción química dio un promedio muestral de 87.3 grados centígrados. ¿Cuál es el promedio muestral en grados Fahrenheit?

In [8]:
farenheit = 87.3
celsius = (farenheit - 32)*5/9
print(f'{round(celsius, 2)}°C')

30.72°C


5.Veintiuna personas tienen una estatura promedio de 168 centímetros. Si se considera la información de una persona adicional, ¿cuál es la estatura que debe tener esta persona para que la estatura promedio se incremente dos centímetros?

*n_personas $(n)$ = $21$* $\\$
*promedio $(\bar{x})$ = $168$* $\\$
*estatura_adicional = $h$* $\\$
*promedio_nuevo $(\bar{x}_f)$ = $\bar{x} + 2$* $\\$

$$
\begin{align*}
&\bar{x}_f = \frac{1}{n+1} * ((n * \bar{x})+h) \\
&h = \left(\bar{x}_f * \left(n+1\right) \right) - \left(n * \bar{x}\right) \\
&= \left(\left(168+2\right) * \left(21+1\right) \right) - \left(21 * 168\right) \\
&= \left(170 * 22 \right) - 3.528 \\
&= 3.740 - 3.528 \\
&= 212_{cm} \\
&\text{In code would look like:} \\ \\
&h= \left(\bar{x}_f * \left(n+1\right) \right) - \left(n * \bar{x}\right) \\
\end{align*}
$$

Se halló la diferencia de las sumas de los dos promedios (muestra de $21$ personas y muestra de $22$ personas) dándonos como resultado $212_{cm}$.


In [9]:
n = 21
mean = 168
mean_f = 168+2
h = (mean_f*(n+1))-(n*mean)
h

212

6.Con la información dada en la siguiente tabla, calcular la calificación de un estudiante en la última unidad de la asignatura considerando que quiere aprobar la materia con 3.2.

In [10]:
# Percentiles
v = np.array([0.2, 0.25, 0.2, 0.15])
x = np.array([3.7, 2.4, 3.0, 3.2])
mean = 3.2 

El valor de la calificación de la última unidad es:
$$
\begin{align*}
&\bar{x} = \frac{\sum_{i=1}^{n} x_iv_i}{\sum_{i=1}^{n} v_i} \Rightarrow
\bar{x} = \frac{3.7 * 0.2 + 2.4 * 0.25 + 3.0 * 0.2 + 3.2 * 0.15 + x_n * 0.2}{0.2 + 0.25 + 0.2 + 0.15 + 0.2} \\
&x_n = \frac{\left(\bar{x} * \left(0.2 + 0.25 + 0.2 + 0.15 + 0.2\right) \right) - \left(3.7 * 0.2 + 2.4 * 0.25 + 3.0 * 0.2 + 3.2 * 0.15\right)}{0.2} \\
&= \frac{\left(\bar{x} * 1 \right) - \left(0.74 + 0.6 + 0.6 + 0.48\right)}{0.2} \\
&= \frac{\bar{x} - 2.42}{0.2} \\
&= \frac{3.2 - 2.42}{0.2} \\
&= \frac{0.78}{0.2} \\
&= 3.9 \\ \\
&\text{In code would look like:} \\ \\
&x_n= \left(\left(\bar{x} * sum\left(v\right)\right) - sum\left(x*v\right)\right)/0.2
\end{align*}
$$

In [11]:
# Appling the formula
l_p = 1-sum(v) # Last percentile value
x_n = ((mean*(sum(v)+l_p)) - sum(x*v))/(l_p)
round(x_n, 2)

3.9

7.Un grupo de 200 estudiantes, cuya estatura promedio es de $160.96$ centímetros, se divide en dos grupos, uno con una estatura promedio de $163.4$ centímetros y otro, con una de $157.3$ centímetros. ¿Cuántos estudiantes tiene cada grupo?

Tenemos: $\\$
$\bar{x} = 160.96 \\ \bar{x}_1 = 163.4 \\ \bar{x}_2 = 157.3 \\ n = 200 \\ n_1 = x \\ n_2 = y \\$
Desarrollamos: $\\$
$$
\begin{align*}
\text{Sabemos que:} \\
&\bar{x}_1*n_1 + \bar{x}_2*n_2 =  \sum_{i=1}^n x_i\\
\text{Así que calculamos: } \sum_{i=1}^n x_i \\
&\sum_{i=1}^n x_i = \bar{x} * n \\
&= 160.96 * 200 \\
&= 32192 \\
\text{Por lo tanto, tenemos: } \\
&\bar{x}_1*n_1 + \bar{x}_2*n_2 = 32192 \hspace{2cm} \text{(1)}\\
\text{Y también: } \\
&n_1 + n_2 = n \\
&n_1 + n_2 = 200 \hspace{2cm} \text{(2)}\\
&n_2 = 200 - n_1 \hspace{2cm} \text{(3)} \\
\text{Reemplazamos para calcular } n_1 \text{:} \\
&\bar{x}_1n_1 + \bar{x}_2*\left(200 - n_1\right) = 32192 \hspace{0.5cm} \text{[ Reemplazamos en la ecuación (1) con la ecuación (3) ]} \\
&\bar{x}_1n_1 + 200\bar{x}_2 - \bar{x}_2n_1 = 32192 \hspace{0.5cm} \text{[ Distibumos ]} \\
&\bar{x}_1n_1 - \bar{x}_2n_1 = 32192 - 200\bar{x}_2 \hspace{0.5cm} \text{[ Pasamos } 200\bar{x}_2 \text{ al otro lado ]} \\
&163.4n_1 - 157.3n_1 = 32192 - (200 * 157.3) \hspace{0.5cm} \text{[ Reemplazamos ]} \\
&6.1n_1 = 32192 - 31460 \\
&6.1n_1 = 732 \\
&n_1 = \frac{732}{6.1} \\
&n= 120 \\
\text{Ahora calculamos } n_2 \text{:} \\
&n_2 = 200 - n_1 \hspace{0.5cm} \text{[ Tomamos la ecuación (3) ]} \\
&n_2= 200 - 120 \hspace{0.5cm} \text{[ Reemplazamos } n_1 \text{ ]} \\
&n_2= 80 \\
\end{align*}
$$

De modo que el primer grupo tiene 120 estudiantes y el segundo, 80.

In [25]:
# In code
mean = 160.96
mean_1 = 163.4
mean_2 = 157.3
n = 200
items_sum = mean * n
# Calculate n_1 with linear algebra formula using the ecuations (1) and (2)
matrix = np.array([[1, 1], 
				[mean_1, mean_2]])
n_1, n_2 = np.linalg.solve(matrix, [n, items_sum])
print(f'{round(n_1)} items in the first group')
print(f'{round(n_2)} items in the second group')

120 items in the first group
80 items in the second group


8.A continuación se presentan los salarios mensuales en miles de pesos, pagados por una empresa del Estado a su personal. En la empresa se presenta un conflicto laboral. El gerente propone un aumento del $2\%$ para cada uno de los empleados, mientras que la junta directiva propone un aumento neto de 10 mil pesos mensuales.

Con base en el promedio, ¿qué es más ventajoso para el Estado y qué para cada grupo de empleados?

In [13]:
# Salarios de los empleados
yj = np.array([300, 400, 500, 600, 700])
# Empleados
nj = np.array([10, 16, 35, 26, 13])
# Numero total de empleados
n = sum(nj)
# Promedio
mean = sum(nj*yj)/n
# Aumentar 2%
op1 = mean*1.02
# Aumentar 10 mil pesos
op2 = mean+10

print(f'Op1: {op1} mil pesos\nOp2: {op2} mil pesos')

Op1: 526.32 mil pesos
Op2: 526.0 mil pesos


Por lo tanto al empleado le sale más comveniente un aumento del $2\%$ pues ganarían $\$526.320$ en promedio. Mientras que con la opción 2 ganarían $\$526.000$ lo cual le conviene más al estado.

9.Muestre que si $k_1$ y $k_2$ son constantes, entonces $M(k1+k2x)=k1+k2M(x)$. $\\$

Demostración\\\\: $\\$
$$
\begin{align*}
&M(k_1+k_2x) \\
&M(k_1+k_2x) = \frac{1}{n} \sum_{i=1}^n k_1 + k_2x_i \hspace{0.5cm} \text{[Por definición de promedio]}\\
&= \frac{1}{n} \left(\sum_{i=1}^n k_1 + \sum_{i=1}^n k_2x_i\right) \hspace{0.5cm} \text{[Distribuimos de la sumatoria]}\\
&= \frac{1}{n} \left(k_1\sum_{i=1}^n + k_2\sum_{i=1}^n x_i \hspace{0.5cm}\right) \text{[Sacamos las constantes de las sumatorias]}\\
&= \frac{1}{n} \left(k_1*n + k_2\sum_{i=1}^n x_i\right) \hspace{0.5cm} \text{[Resolvemos la primera sumatoria (Sumatoria de 1 a n de 1 es n)]}\\
&= k_1 + k_2\frac{1}{n}\sum_{i=1}^n x_i \hspace{0.5cm} \text{[Distribuimos 1/n (en el primer sumando se cancela la n)]}\\
&= k_1 + k_2\bar{x} \hspace{0.5cm} \text{[Reemplazamos }\frac{1}{n}\sum_{i=1}^n x_i = \bar{x}\text{]}\\
&M(k_1+k_2x) = k_1 + k_2M(x) \hspace{0.5cm} \text{[Reemplazamos }\bar{x} = M(x)\text{]}\\
\end{align*}
$$

10.Considere un conjunto de datos particionados en $m$ grupos con tamaños $n_1,\dots,n_m$ y medias $\bar{x}_1,\dots,\bar{x}_m$, tales que $n_1=\dots=n_m=c$. Muestre que si $n=c\cdot m$, entonces la media de todo el conjunto de datos $\bar{x}$ coincide con la media de las medias de los grupos $\bar{\bar{x}}$.

In [14]:
# Pendiente